documentation/behind-the-scenes-of-adding-a-file.md

```diff
@@ -0,0 +1,233 @@
+## Purpose
+
+The steps outlined below are taken to "add a file with Filecoin Pin".  This document is intended to provide more info about what happens "behind the scenes" as it uses underlying libraries like [`synapse`](glossary.md#synapse) and the [Filecoin Onchain Cloud](glossary.md#filecoin-onchain-cloud) offering.
+
+## Diagram
+
+_Key_ 
+Blue | Orange | FP | SP
+-- | -- | -- | --
+non-blockchain step | blockchain step | Filecoin Pin | Storage Provider
+
+
+```mermaid
+graph TD
+
+    %% Node Definitions
+    %% Non-blockchain nodes
+    Start([User: Select File to Add])
```

Comment:
is this just for the website? because this could be "Select File or Directory to Add"